src/libPMacc/include/traits/GetUniqueTypeId.hpp

```diff
@@ -78,21 +85,9 @@ struct GetUniqueTypeId
      */
     static const ResultType uid(uint64_t maxValue = (uint64_t(1) << (sizeof (ResultType) * CHAR_BIT)) - uint64_t(1))
```

Comment:
- `std::numeric_limits` is not defined for all types in C++98 (e.g. 64bit)
- `~uint64_t(0)` is a good idea, I will check if it is possible to use it